spinalcordtoolbox/reports/qc.py

```diff
@@ -264,17 +278,20 @@ def wrapped_f(sct_slice, *args):
             """
             self.qc_report.slice_name = sct_slice.get_name()
 
-            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
-            aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
-
             self.qc_report.make_content_path()
             logger.info('QcImage: %s with %s slice', func.__name__, sct_slice.get_name())
 
-            if self._angle_line is None:
-                img, mask = func(sct_slice, *args)
-            else:
+            if self._angle_line is not None:
                 [img, mask], centermass = func(sct_slice, *args)
                 self._centermass = centermass
+            if self._fps is not None:
+                [images_after_moco, images_before_moco], centermass = func(sct_slice, *args)
+                self._centermass = centermass
+            else:
+                img, mask = func(sct_slice, *args)
+
+            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
+            self.aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
```

Comment:
Hmm... looking at the `__call__` function, it's quite complex right now. It has a lot of interwoven branching paths based on various conditions. By adding moco code using `if self._fps is not None` patches, the function becomes even more complex.

To me, this is a ["code smell"](https://en.wikipedia.org/wiki/Code_smell) indicating that `__call__` should be refactored in a more [modular](https://aist.global/en/importance-of-modular-programming) way.

Because it isn't very modular, I think `__call__` is quite error-prone and hard to maintain. For example, what happens if the `fps` parameter got passed to a non-moco instance of `QcImage`? I have a feeling `__call__` would break.

Reply:
<details>
<summary> Aside about <code>__call__</code> </summary>

Looking at the `__call__` function, it's quite complex right now. It has a lot of interwoven branching paths based on various conditions. By adding moco code using `if self._fps is not None` patches, the function becomes even more complex.

To me, this is a ["code smell"](https://en.wikipedia.org/wiki/Code_smell) indicating that `__call__` should be refactored in a more [modular](https://aist.global/en/importance-of-modular-programming) way. Because it isn't very modular, I think `__call__` is quite error-prone and hard to maintain right now.
</details>